### PR TITLE
Add stage all/unstage all mappings

### PR DESCRIPTION
**Pull request by mg979**

### CODE REVIEW

File: plugin/magit.vim [hunk 1]
```diff
@@ -1256,7 +1256,7 @@ function! magit#jump_hunk(dir)
 				break
 			endtry
 		endwhile
-		silent execute "normal! zt"
+		silent execute "normal! zt2\<C-Y>"
```
Comment (mg979) [inline plugin/magit.vim]: This helps seeing what file are you staging, and when you move to a different file with `C-N`/`C-P`.

File: autoload/magit/mapping.vim [hunk 2]
```diff
@@ -2,6 +2,8 @@
 let g:magit_stage_file_mapping     = get(g:, 'magit_stage_file_mapping',        'F' )
 let g:magit_stage_hunk_mapping     = get(g:, 'magit_stage_hunk_mapping',        'S' )
 let g:magit_stage_line_mapping     = get(g:, 'magit_stage_line_mapping',        'L' )
+let g:magit_stage_all_mapping      = get(g:, 'magit_stage_all_mapping',         'gA' )
+let g:magit_unstage_all_mapping    = get(g:, 'magit_unstage_all_mapping',       'gU' )
```
Comment (mg979) [inline autoload/magit/mapping.vim:6]: I'm using `gA` and `gU`, but they can be obviously changed (gS/gR, for example). `gR` could be used in the future for remotes, so I wouldn't use it.

File: plugin/magit.vim [hunk 3]
```diff
@@ -1116,6 +1116,18 @@ function! magit#stage_hunk(discard)
...
+  call system('git add -u')
+  call magit#update_buffer()
+endfunction
+
+" magit#unstage_all
+function! magit#unstage_all()
+  call system('git reset --mixed')
+  call magit#update_buffer()
+endfunction
+
```
Comment (mg979) [inline plugin/magit.vim]: The functions are barebones, but they work, assuming the pwd hasn't changed. I don't know how you would do it but this works for me.

**Comment by jreybert** [comment]
Thanks @mg979 for this pull request.

Several remarks:
* try to do one pull request per feature
* in commit, please, do not mix feature and esthetic changes (in 2-line space commit, one diff line is relevant for the feature, and 4 lines remove end of line spaces). I am ashamed of these end of line spaces, but keep this correction in a separate commit.

Regarding the 2-line space commit, I don't have this problem because I set the option `scrolloff` to 5 by default in my vimrc. However, this is a good point that not everyone set this option, and it helps a lot in vimagit case.

I prefer to rely on the option `scrolloff` instead of hand crafted solution. I pushed 6a66b2f8d in branch next for this new feature.

About the push all feature, it is interesting, but:
* I am still not sure about the default mapping
* please use this king of command to start a git command

```
magit#sys#system(g:magit_git_cmd . " add -u")
```

**Comment by mg979** [comment]
> * try to do one pull request per feature
> * in commit, please, do not mix feature and esthetic changes (in 2-line space commit, one diff line is relevant for the feature, and 4 lines remove end of line spaces). I am ashamed of these end of line spaces, but keep this correction in a separate commit.

Sure, I removed that commit (I didn't think about `scrolloff`, much better solution), those spaces still get a bit in the way though.

About the mappings, I think it's better to use `g` for 'lesser commands'. It's how vim does it after all. I'd be against mapping `A` because it would hurt commit message editing (`C` already does but it's less important than `A`). Maybe `gSA` and `gUA`? You choose anyway.

**Comment by mckellygit** [comment]
How about a single routine and mapping that stages/unstages ?
As in -
```
function! magit#stage_all()
    let section=magit#helper#get_section()
    if ( section == 'unstaged' )
        call magit#sys#system(g:magit_git_cmd . " add -u")
    elseif ( section == 'staged' )
        call magit#sys#system(g:magit_git_cmd . " reset --mixed")
    endif
    call magit#update_buffer()
```
I've been playing around with this and thinking of adding a popup diaglog or preview/quickfix to display any warnings/errors from the system call.

**Pull request closed without merging**